### PR TITLE
Update to latest minecraft version and lastest forge gradle

### PR DESCRIPTION
**Pull request by Exopandora**
This PR updates the forge gradle to 3.0.184 and fixes some issues coming with that:
- Updated dependencies
- Fixed eclipse runs
- Fixed createRelease:reobf task for new fg versions
- Fixed limitation to java 8 

This PR also updates the conf/settings.json to minecraft 1.16.2 and the examplemod sources and patches

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -54,36 +61,47 @@ project(':mcp') {
...
     }
     task runclient(type: JavaExec) {
         doFirst {
             mkdir 'runclient'
         }
         classpath sourceSets.main.runtimeClasspath
-        args = ['--accessToken', '0', '--version', '1.13']
-        main 'net.minecraft.client.main.Main'
+        args = ['--accessToken', '0', '--version', '1.16.2']
+        main settings.servermain
```
Comment (Earthcomputer) [inline build.gradle:91]: `settings.servermain` on the client?

File: patches/net/minecraft/command/impl/CloneCommand.java.patch [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
+@@ -1,5 +1,6 @@
+ package net.minecraft.command.impl;
  
 +import com.example.mod.FillLimitCommand;
-+
- public class CloneCommand
- {
-     private static final SimpleCommandExceptionType OVERLAP_EXCEPTION = new SimpleCommandExceptionType(new TranslationTextComponent("commands.clone.overlap"));
-@@ -109,9 +111,9 @@
+ import com.google.common.collect.Lists;
```
Comment (Earthcomputer) [inline patches/net/minecraft/command/impl/CloneCommand.java.patch:7]: Imports should go below the vanilla ones

File: patches/net/minecraft/command/impl/FillCommand.java.patch [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
+@@ -1,5 +1,6 @@
+ package net.minecraft.command.impl;
  
 +import com.example.mod.FillLimitCommand;
-+
- public class FillCommand
- {
-     private static final Dynamic2CommandExceptionType TOO_BIG_EXCEPTION = new Dynamic2CommandExceptionType((p_208897_0_, p_208897_1_) ->
-@@ -70,9 +72,9 @@
+ import com.google.common.collect.Lists;
```
Comment (Earthcomputer) [inline patches/net/minecraft/command/impl/FillCommand.java.patch:7]: Import order

**Review by Earthcomputer — COMMENTED** [review]
Overall looks good, I'll need to test it but ya

**Comment by Exopandora** [comment]
Note: I don't know if idea runs generate correctly as i use eclipse

**Pull request merged**